### PR TITLE
Update resource_provisioner.go

### PR DESCRIPTION
**Pull request by gibonurs**
This patch allow to correctly update chef-vaults when chef node needs to be recreated. If client with old key exists in chef-vault it will try to remove it first and then will add node with new key

### CODE REVIEW

**Comment by krzkowalczyk** [comment]
I would change it to bug, as with chef-vault 3.1.0 version existing client keys are not updated, so recreating a node in terraform does not refresh vaults. It's needed to remove client first.

**Comment by jmccann** [comment]
@svanharmelen I know you helped me to get chef vault bootstrapping as part of chef provisioner.  Just ran into this issue myself.  Was wondering your thoughts on it.  Thanks!

File: builtin/provisioners/chef/resource_provisioner.go [hunk 1]
```diff
@@ -560,6 +560,18 @@ func (p *provisioner) configureVaultsFunc(gemCmd string, knifeCmd string, confDi
 
 		for vault, items := range p.Vaults {
 			for _, item := range items {
+
```
Comment (svanharmelen) [inline builtin/provisioners/chef/resource_provisioner.go]: _nitting_ Could you delete this empty line?

File: builtin/provisioners/chef/resource_provisioner.go [hunk 2]
```diff
@@ -560,6 +560,18 @@ func (p *provisioner) configureVaultsFunc(gemCmd string, knifeCmd string, confDi
...
 			for _, item := range items {
+				
+				removeCmd := fmt.Sprintf("%s vault remove %s %s -C %s -M client %s",
+					knifeCmd,
+					vault,
+					item,
+					p.NodeName,
+					options,
+				)
+				if err := p.runCommand(o, comm, removeCmd); err != nil {
```
Comment (svanharmelen) [inline builtin/provisioners/chef/resource_provisioner.go]: Are you sure this doesn't give an error when the client doesn't exist?

Comment (krzkowalczyk) [inline builtin/provisioners/chef/resource_provisioner.go]: I confirm, there is no error when client doesn't exist. At least in chef-vault <= 3.2.0.
IMHO chef-vault gem version should be pinned to ensure compatibility.

**Comment by DukeXar** [comment]
Would be nice to provide an ability to specify version of the chef-client
actually.

**Comment by svanharmelen** [comment]
@DukeXar you already can specify a Chef-Client version: https://www.terraform.io/docs/provisioners/chef.html#version-string-

**Comment by mildwonkey** [comment]
Hello! I am going to close this older PR due to inactivity. Please do open a new PR if this is something you are still interested in working on. Thanks!

**Pull request closed without merging**